tiledb/sm/cpp_api/profile_experimental.h

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file   profile_experimental.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2025 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the experimental profile C++ API.
+ */
+
+#ifndef TILEDB_CPP_PROFILE_EXPERIMENTAL_H
+#define TILEDB_CPP_PROFILE_EXPERIMENTAL_H
+
+#include "context.h"
+#include "deleter.h"
+#include "exception.h"
+#include "tiledb_experimental.h"
+
+#include <memory>
+#include <string>
+
+namespace tiledb {
+
+class ProfileException : public TileDBError {
+ public:
+  explicit ProfileException(const std::string& msg)
+      : TileDBError(std::string("[TileDB::C++ API][Profile]: " + msg)) {
+  }
+};
+
+class Profile {
+ public:
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+
+  /** Default constructor. */
+  Profile()
+      : Profile(nullptr, nullptr) {
+  }
+
+  /**
+   * Constructor.
+   *
+   * @note Use of `homedir` is intended primarily for testing purposes, to
+   * preserve local files from in-test changes. Users may pass their own
+   * `homedir`, but are encouraged to use `nullptr`, the default case.
+   *
+   * @param name The profile name if you want to override the default.
+   * @param homedir The local $HOME directory path if you want to override the
+   * default.
+   */
+  explicit Profile(
+      std::optional<std::string> name = std::nullopt,
+      std::optional<std::string> homedir = std::nullopt) {
+    const char* n = name.has_value() ? name->c_str() : nullptr;
```

Comment:
Ah, the flaw of `std::string_view`, that it cannot guarantee for you a NULL-terminated string.

Reply:
Yep, exactly - using `std::string` here ensures `.c_str()` gives a null-terminated `const char*`.

Reply:
Not relevant anymore.